src/core/namespace.c

```diff
@@ -1346,14 +1346,68 @@ static int mount_mqueuefs(const MountEntry *m) {
         return 1;
 }
 
+static VeritySettings* verities_find_by_roothash(
+                VeritySettings *haystack,
+                size_t n_haystack,
+                const VeritySettings *needle) {
+
+        assert(haystack || n_haystack == 0);
+        assert(needle);
+
+        FOREACH_ARRAY(v, haystack, n_haystack)
+                if (memcmp_nn(v->root_hash,
+                              v->root_hash_size,
+                              needle->root_hash,
+                              needle->root_hash_size) == 0)
+                        return v;
+
+        return NULL;
+}
+
+static int verities_find_duplicates(
+                const char *path,
+                VeritySettings *verities,
+                size_t n_verities,
+                VeritySettings *ret_verity) {
+
+        _cleanup_(verity_settings_done) VeritySettings verity = VERITY_SETTINGS_DEFAULT;
+        int r;
+
+        assert(path);
+        assert(verities || n_verities == 0);
+        assert(ret_verity);
+
+        r = verity_settings_load(&verity, path, /* root_hash_path= */ NULL, /* root_hash_sig_path= */ NULL);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to check verity root hash of %s: %m", path);
+
+        if (verity.root_hash_size == 0) {
+                *ret_verity = (VeritySettings) VERITY_SETTINGS_DEFAULT;
```

Comment:
hmm, we might have loaded other metadata, no? hence no need to flush this out here? you could as well return that as in in the other case, no?